25.7.2 patch for shadowsocks

eguun opened the pull request:
Patch aimed to fix Shadowsocks plugin pivot from libev to rust

Issues on 25.7.2 are documented here: https://forum.opnsense.org/index.php?topic=48568.0
Github.com ask to pivot from libev to rust documented here: https://github.com/opnsense/plugins/issues/4662

The rust build run fine (no crash/Segfault....), but isn't integrated into OPNSense GUI to start/stop/monitor deamon, as the OPNSense config is still looking for the libev version of files (config, .pid ...)

This PR here is aimed to mirror Slashing updates from opnsense forums.
Slashing has built a .zip with manual files updates and identified some files to fix and properly link the OPNSense GUI to the updated rust files.


What this PR does not do vs Slashing's updates:
delete files:
```
/etc/rc.conf.d/shadowsocks_libev
/etc/rc.conf.d/ss_local
```
Because I couldn't find where/how to mark those files for deletion

My first PR - feedbacks welcome

fichtner commented on net/shadowsocks/src/etc/rc.d/sslocal-rust:
```diff
@@ -0,0 +1,33 @@
+#!/bin/sh
```
aren't these files installed by default?

eguun replied:
Thanks indeed these files will be created
I don't see any action item on the github.com platform, so I'll mark this as resolved.

fichtner commented on net/shadowsocks/src/etc/rc.d/ssserver-rust:
```diff
@@ -0,0 +1,33 @@
+#!/bin/sh
```
same here

fichtner commented on net/shadowsocks/src/opnsense/service/conf/actions.d/actions_shadowsocks.conf:
```diff
@@ -1,47 +1,47 @@
 
 [status]
-command:/usr/local/etc/rc.d/shadowsocks_libev status;exit 0
+command:/usr/local/etc/rc.d/ssserver-rust status;exit 0
 parameters:
```
```suggestion
command:/usr/local/etc/rc.d/ssserver-rust status
errors:no
parameters:
```

fichtner commented:
Removing old config files is irrelevant as shadowsocks-libev is going to be removed so nothing can start it.

kvoffka commented:
Can I ask a weird question?
I use a cipher like 2022-blake3-aes 128-gcm.
Maybe you should add a field for manual cipher entry?

eguun commented:
> Can I ask a weird question? I use a cipher like 2022-blake3-aes 128-gcm. Maybe you should add a field for manual cipher entry?

Maybe not the best place where to ask for feature enhancement:
- this PR is to hotfix the insertion of Shadowsocks-rust within OPNSense (which used to use the deprecated -libev version of the port)
- not aimed to enhance the feature-set, but make the basic works as OPNsense will use 1:1 the port of FreeBSD
- feature request on ShadowSocks as you suggest will get a higher chance of success when pushed there: https://www.freshports.org/net/shadowsocks-rust

Regards

fichtner commented on net/shadowsocks/src/opnsense/service/templates/OPNsense/Shadowsocks/sslocal-rust:
```diff
@@ -1,6 +1,6 @@
 {% if helpers.exists('OPNsense.shadowsocks.local.enabled') and OPNsense.shadowsocks.local.enabled == '1' %}
-ss_local_enable="YES"
-ss_local_flags=""
+sslocal_rust_enable="YES"
+sslocal_rust_flags=""
```
it also looks like we don't need sslocal_rust_flags anymore

eguun replied:
Indeed - I appreciate this simple & minimalism approach

fichtner commented on net/shadowsocks/src/opnsense/service/templates/OPNsense/Shadowsocks/ssserver-rust:
```diff
@@ -1,6 +1,6 @@
-shadowsocks_libev_enable="YES"
-shadowsocks_libev_flags=""
+ssserver_rust_enable="YES"
+ssserver_rust_flags=""
```
```suggestion
```

fichtner commented:
I'll merge this now and do the rest of the tweaking in the repo. Thanks!

eguun commented:
> I'll merge this now and do the rest of the tweaking in the repo. Thanks!

Great to read. 
Thanks to you and kudos to your reactivity and support!

fichtner commented:
https://github.com/opnsense/plugins/commit/0354e1851 resembles the state that's on master now and I will hotfix when you tell me it works (it did create config and start here in my tests)

eguun commented:
> [0354e1851](https://github.com/opnsense/plugins/commit/0354e1851) resembles the state that's on master now and I will hotfix when you tell me it works (it did create config and start here in my tests)

Thanks @fichtner - I confirm that what we implemented in the PR does work for both aarch64 and x64
When hotfix rolled out, I will test on those setup by removing plugin and installing it again

fichtner commented:
@eguun hotfix has been published

eguun commented:
> @eguun hotfix has been published

I confirm it works, thanks!

Details:
- on AMD64, flipped setting from "community" to "devel"
- updated to Version 26.1.a_200 
- deleted shadowsocks plugin
- removed leftover files related to shadowsocks, specifically:
rm /etc/rc.conf.d/ss*
rm -rf /usr/local/opnsense/service/templates/OPNsense/Shadowsocks/
- re-installed shadowsocks plugin (the new version 1.2_1)
- success criterias: deamon is running and shows as running in OPNSense dashboard web UI, remote connection successful

eguun commented:
One small typo observed:

in plugin description for version 1.2 in changelog
"Fix compatiblity of plugin" - there's one "i" missing in "compatibility"

It's a detail, does not prevent either run of plugin or readability/understandability.
Non blocking to move from dev to prod

eguun commented:
> Ok, fixed in [940d34b](https://github.com/opnsense/plugins/commit/940d34bddaf96d81849d13382d2c5b1883e0c89e), nothing that needs hotfixing at the moment. Thanks for your help ❤️

Kudos are to you for your guidance and hand helding, thanks!
I started this with PR = Press Release :)
a nice journey packed with learnings.

Process-wise, when would it be realistic to expect this hotfix make it to production?
